src/libraries/System.Linq/src/System/Linq/DefaultIfEmpty.SpeedOpt.cs

```diff
@@ -18,10 +18,21 @@ public TSource[] ToArray()
 
             public List<TSource> ToList()
             {
-                List<TSource> list = _source.ToList();
-                if (list.Count == 0)
+                List<TSource> list;
+                if (_source is IIListProvider<TSource> listProvider && listProvider.GetCount(onlyIfCheap: true) == 0)
```

Comment:
This doesn't show up as a regression when, for example, GetCount returns 1?

Reply:
Reverted due to the other comment.